exec/cnex/cnex.c

```diff
@@ -1310,9 +1364,54 @@ void exec_JUMPTBL(TExecutor *self)
     }
 }
 
-void exec_CALLX(void)
+void exec_CALLX(TExecutor *self)
 {
-    fatal_error("exec_CALLX not implemented");
+    self->ip++;
+    uint32_t mod = exec_getOperand(self);
+    uint32_t name = exec_getOperand(self);
+    uint32_t out_param_count = exec_getOperand(self);
+
+    char modname[256];
+    snprintf(modname, 256, "%s/%sneon_%s", path_getPathOnly(self->module->source_path), LIBRARY_NAME_PREFIX, self->module->bytecode->strings[mod]->data);
+    TExtensionModule *exmod = ext_findModule(modname);
+    if (exmod == NULL) {
+        void_function_t init = rtl_foreign_function(self, modname, "Ne_INIT");
+        if (init == NULL) {
+            fatal_error("neon_exec: function Ne_INIT not found in %s\n", modname);
+        }
+        (init)(&ExtensionMethodTable);
+    }
+
+    void_function_t p;
+    char funcname[256];
+    snprintf(funcname, 256, "%s%s", "Ne_", self->module->bytecode->strings[name]->data);
+    p = rtl_foreign_function(self, modname, funcname);
```

Comment:
There should at least be a TODO comment here to make this a bit more efficient. You don't actually have to call `rtl_foreign_function` on every run of this opcode.

Reply:
Added comment for ToDo.  This was something I had totally forgot about adding.